KarhooUISDK/UIComponents/LegalNotice/KarhooLegalNoticeView.swift

```diff
@@ -0,0 +1,135 @@
+        }()
+    
+    private var attributedLabel: UILabel = {
+        let legalNoticeLabel = UILabel()
+        legalNoticeLabel.translatesAutoresizingMaskIntoConstraints = false
+        legalNoticeLabel.accessibilityIdentifier = KHLegalNoticeViewID.text
+        legalNoticeLabel.font = KarhooUI.fonts.captionRegular()
+        legalNoticeLabel.textAlignment = .justified
+        legalNoticeLabel.attributedText = LegalNoticeStringBuilder().legalNotice()
+        legalNoticeLabel.numberOfLines = 0
+        legalNoticeLabel.isUserInteractionEnabled = true
+        return legalNoticeLabel
+    }()
+    
+    // MARK: - Init
+    init(parent: UIViewController, linkParser: LinkParser = LinkParser()) {
+        zeroHeightTextConstreint = attributedLabel.heightAnchor.constraint(equalToConstant: 0)
+        self.parentViewController = parent
+        self.legalNoticeMailComposer = KarhooLegalNoticeMailComposer(parent: parentViewController)
+        super.init(frame: .zero)
+        self.setUpView()
+    }
+    
+    required init(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // MARK: - Setup
+    private func setUpView() {
+        accessibilityIdentifier = KHLegalNoticeViewID.view
+        translatesAutoresizingMaskIntoConstraints = false
+        
+        attributedLabel.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.singleTap(_:))))
+        
+        addSubview(legalNoticeButton)
+        addSubview(attributedLabel)
+        hideLegalNoticePressed()
+    }
+    
+    override func updateConstraints() {
+        if !didSetUpConstraints {
+            legalNoticeButton.anchor(
+                top: topAnchor,
+                leading: leadingAnchor,
+                bottom: attributedLabel.topAnchor,
+                paddingLeft: 8.0,
+                paddingRight: 8.0)
+            
+            attributedLabel.anchor(
+                top: legalNoticeButton.bottomAnchor,
+                leading: leadingAnchor,
+                bottom: bottomAnchor,
+                trailing: trailingAnchor,
+                paddingLeft: 8.0,
+                paddingBottom: 16,
+                paddingRight: 8.0)
+            
+            didSetUpConstraints.toggle()
+        }
+        super.updateConstraints()
+    }
+    
+    @objc func singleTap(_ tap : UITapGestureRecognizer) {
+           let attributedText = NSMutableAttributedString(attributedString: self.attributedLabel.attributedText!)
+        attributedText.addAttributes([NSAttributedString.Key.font: self.attributedLabel.font as UIFont], range: NSMakeRange(0, (self.attributedLabel.attributedText?.string.count)!))
+
+           // Create instances of NSLayoutManager, NSTextContainer and NSTextStorage
+           let layoutManager = NSLayoutManager()
+           let textContainer = NSTextContainer(size: CGSize(width: (self.attributedLabel.frame.width), height: (self.attributedLabel.frame.height)+100))
```

Comment:
1. For clarity, consider to break the line into several lines.
2. `width: (self.attributedLabel.frame.width)` ← the brackets are discouraged 
3. Same for `(self.attributedLabel.frame.height)+100)`
4. It is suggested to add whitespace before and after operator (`+` in this case)